app/code/Magento/Cms/Helper/Wysiwyg/Images.php

```diff
@@ -242,12 +242,16 @@ public function getCurrentPath()
                 }
             }
             try {
-                $currentDir = $this->_directory->getRelativePath($currentPath);
-                if (!$this->_directory->isExist($currentDir)) {
-                    $this->_directory->create($currentDir);
+                $currentTreePath = $this->_getRequest()->getParam('current_tree_path');
+                if ($currentTreePath) {
+                    $currentTreePath = $this->convertIdToPath($currentTreePath);
+                    $currentDir = $this->_directory->getRelativePath($currentTreePath);
+                    if (!$this->_directory->isExist($currentDir)) {
+                        $this->_directory->create($currentDir);
```

Comment:
Thanks for your concern for security (as mentioned in issue description). I tried to find any security issue with this but couldn't find anything yet. I'm successfully getting "Path is Invalid" when attempting with malicious dir name.